espnet/nets/pytorch_backend/e2e_asr_maskctc.py

```diff
@@ -0,0 +1,216 @@
+# Copyright 2020 Johns Hopkins University (Shinji Watanabe)
+#                Waseda University (Yosuke Higuchi)
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+"""
+Mask CTC based non-autoregressive speech recognition model (pytorch).
+See https://arxiv.org/abs/2005.08700 for the detail.
+
+"""
+import logging
+import math
+from itertools import groupby
+
+import numpy
+import torch
+
+from espnet.nets.pytorch_backend.e2e_asr_transformer import E2E as E2ETransformer
+from espnet.nets.pytorch_backend.e2e_asr import CTC_LOSS_THRESHOLD
+from espnet.nets.pytorch_backend.nets_utils import make_non_pad_mask
+from espnet.nets.pytorch_backend.nets_utils import th_accuracy
+from espnet.nets.pytorch_backend.maskctc.add_mask import mask_uniform
+
+
+class E2E(E2ETransformer):
+    """E2E module.
+
+    :param int idim: dimension of inputs
+    :param int odim: dimension of outputs
+    :param Namespace args: argument Namespace containing options
+
+    """
+
+    @staticmethod
+    def add_arguments(parser):
+        """Add arguments."""
+        E2ETransformer.add_arguments(parser)
+        E2E.add_maskctc_arguments(parser)
+
+        return parser
+
+    @staticmethod
+    def add_maskctc_arguments(parser):
```

Comment:
No additional argument now?

Reply:
No for now, should I remove this method?

Reply:
Will you have something in the near future?
I'm fine to keep it for such a future preparation for now.

Reply:
ok, I'll keep it for the future development.